app-modules/engagement/src/Filament/Pages/MessageCenter.php

```diff
@@ -300,6 +305,12 @@ public function applyFilters(Builder $query, string $dateColumn, string $idColum
                         ->open()
                         ->pluck('respondent_id')
                 );
+            })
+            ->when($this->filterMemberOfCareTeam === true, function (Builder $query) use ($idColumn) {
+                $query->whereIn(
+                    $idColumn,
+                    CareTeam::query()->pluck('educatable_id')
```

Comment:
Hmm, this doesn't seem quite right. Not only does it result in querying all care team pivots, but it isn't even scoped to the user.

I am pretty sure you want something move like `$this->user->careTeams()->pluck('educatable_id')`